codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/CborParserGenerator.kt

```diff
@@ -404,9 +435,12 @@ class CborParserGenerator(
                 """
                 pub(crate) fn $fnName(value: &[u8], mut builder: #{Builder}) -> #{Result}<#{Builder}, #{Error}> {
                     #{StructurePairParserFn:W}
-                    
+
                     let decoder = &mut #{Decoder}::new(value);
-                    
+                    if decoder.position() == value.len() {
+                        return #{Ok}(builder);
+                    }
```

Comment:
Isn't this equivalent to `0 == value.len()`?

Is this check necessary?

Reply:
Thanks for catching, client protocol tests used to panic and be unable to decode single bytes without this when my branch was off of an early commit in https://github.com/smithy-lang/smithy-rs/pull/2544, but looks like no longer needed. Removed the check in 21e1ab2.